pymc_marketing/mmm/plot.py

```diff
@@ -368,303 +287,589 @@ def _dim_list_handler(
+
+        Parameters
+        ----------
+        original_scale : bool, default False
+            Whether to plot in original scale (True) or scaled space (False).
+            If True, requires channel_contribution_original_scale in posterior.
+        constant_data : xr.Dataset, optional
+            Dataset containing constant_data group with required variables:
+            - 'channel_data': Channel spend data (dims include "date", "channel")
+            - 'channel_scale': Scaling factor per channel (if original_scale=True)
+            - 'target_scale': Target scaling factor (if original_scale=True)
+
+            If None, uses self.idata.constant_data.
+            This parameter allows:
+            - Testing with mock constant data
+            - Plotting with alternative scaling factors
+            - Comparing different data scenarios
+        posterior_data : xr.Dataset, optional
+            Dataset containing posterior group with channel contribution variables.
+            Must contain 'channel_contribution' or 'channel_contribution_original_scale'.
+            If None, uses self.idata.posterior.
+            This parameter allows:
+            - Testing with mock posterior samples
+            - Plotting external inference results
+            - Comparing different model fits
+        dims : dict[str, str | int | list], optional
+            Dimension filters to apply. Examples:
+            - {"geo": "US"} - Single value
+            - {"geo": ["US", "UK"]} - Multiple values
+
+            If provided, only the selected slice(s) will be plotted.
+        backend : str, optional
+            Plotting backend to use. Options: "matplotlib", "plotly", "bokeh".
+            If None, uses global config via mmm_plot_config["plot.backend"].
+            Default is "matplotlib".
+
+        Returns
+        -------
+        PlotCollection
+            arviz_plots PlotCollection object containing the plot.
+
+            Use ``.show()`` to display or ``.save("filename")`` to save.
+            Unlike the legacy suite which returned ``(Figure, Axes)``,
+            this provides a unified interface across all backends.
+
+        Raises
+        ------
+        ValueError
+            If required data not found in self.idata and not provided explicitly.
+        ValueError
+            If 'channel_data' not found in constant_data.
+        ValueError
+            If original_scale=True but channel_contribution_original_scale not in posterior.
+
+        See Also
+        --------
+        saturation_curves : Add posterior predictive curves to this scatter plot
+        LegacyMMMPlotSuite.saturation_scatterplot : Legacy matplotlib-only implementation
+
+        Notes
+        -----
+        Breaking changes from legacy implementation:
+
+        - Returns PlotCollection instead of (Figure, Axes)
+        - Lost **kwargs for matplotlib customization (use backend-specific methods)
```

Comment:
from what I can see, `kwargs` could only change the figure size through `width_per_col` and `height_per_row`. I would make the message either more specific or more actionable (for example pointing to how one can change the figuresize now, even if matplotlib only example).